### ISSUE TITLE
bug: android back button not working on capacitor 3 with ionic.

### ISSUE THREAD

**Issue by RaphaelWoude** [ionic-team/capacitor#4317]
## Bug Report

I have re-created this issue because there are still issues with the android back button with capacitor 3 and ionic.
Even with the fix that was suggested in the last issue, it did not solve the problem.
Here is the original issue: https://github.com/ionic-team/capacitor/issues/4300

### Capacitor Version

```
Capacitor Doctor   

Latest Dependencies:

  @capacitor/cli: 2.4.7
  @capacitor/core: 2.4.7
  @capacitor/android: 2.4.7
  @capacitor/ios: 2.4.7

Installed Dependencies:

  @capacitor/ios: not installed
  @capacitor/cli: 3.0.0-beta.6
  @capacitor/core: 3.0.0-beta.6
  @capacitor/android: 3.0.0-beta.6
```

### Platform(s)

Android

### Current Behavior

The app closes when tapping the android back button instead of going back to the last route. In this case that's the tab1 route.

### Expected Behavior

The app should go to the last route when tapping the android back button. In this case that's the tab1 route.

### Code Reproduction

1. Clone: https://github.com/RaphaelWoude/Example
2. Run the following commands:
```
npm install
npm run build
npm run sync
npm run copy
npm run android
```
3. Build with android studio.
4. Start the app.
5. Click on the test button.
6. Press the hardware back button.

### Other Technical Details

`npm --version` output:
6.14.10
`node --version` output:
v14.15.4
`pod --version` output (iOS issues only):

### Additional Context

I have provided a v2 version branch of the app in the same repository. This one does exactly what the app should do by going back to the last route. 
Here is also a video running on version 3 that shows that the ionic back button works but as soon as I tap the android back button it closes the app.

https://user-images.githubusercontent.com/47354694/110136780-610f6300-7dd0-11eb-896c-d305e8f65f05.mp4

**Comment by RaphaelWoude** [comment]
I completely forgot to change the MainActivity to the auto register when upgrading to v3 🤦‍♀️
Just tried it out and works fine now. Thank you so much ❤

**Comment by klemensz** [comment]
In our case we had the same issue with the non-functioning back button (which just exited the app). What fixed it was installing the `@capacitor/app` and `@capacitor/haptics` dependencies and then running `npx cap sync`. 

The app was initially set up as a Capacitor 3 (RC0) Ionic app (migrating from Cordova) so it's a bit strange that these plugins are not installed by default.

**Comment by rhyek** [comment]
@klemensz that works nicely. One issue I'm seeing now is that on the initial route, when hitting back it's not closing the app as you'd expect it to.

**Comment by klemensz** [comment]
It's maybe related to this issue? https://github.com/ionic-team/ionic-framework/issues/16611

**Comment by rhyek** [comment]
@klemensz, yeah, thanks. This is documented, actually: https://ionicframework.com/docs/developing/hardware-back-button#exiting-the-app

Seems odd that I have to define that behaviour and I'm not work what implications using App.exitApp() has regarding state. Going to have to research/test that.
